pkg/provider/types/parameters.go

```diff
@@ -0,0 +1,118 @@
+package types
+
+import (
+	"strconv"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+// KeyVaultObject holds keyvault object related config
+type KeyVaultObject struct {
```

Comment:
should we rename constants.go -> `types.go` and put all structs and const in it? For eg https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/api/types.go

I have seen this pattern in other k8s repos as well and personally follow the same.

Reply:
I have no strong preference for this. I was using the kubelet package in k/k as ref which has constants in `constants.go` and we used that in other projects: https://github.com/kubernetes/kubernetes/tree/master/pkg/kubelet/types

I've move the structs to a single file and renamed it to `types.go` here.